### PR TITLE
Expose personnel category type for military service episodes

### PR DESCRIPTION
**Pull request by emersonthis**
Closes [11589](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/11589)

## Background

We want to include the `personnel_category_type_code` attribute of a EMIS::Models::MilitaryServiceEpisode to differentiate between active duty, reserves, etc. This effects the response of `/v0/profile/service_history`

**Before**
```
{
    "data": {
        "id": "",
        "type": "arrays",
        "attributes": {
            "service_history": [
                {
                    "branch_of_service": "Army",
                    "begin_date": "1951-09-04",
                    "end_date": "1957-01-04"
                }
            ]
        }
    }
}
```

**After**
```
{
    "data": {
        "id": "",
        "type": "arrays",
        "attributes": {
            "service_history": [
                {
                    "branch_of_service": "Army",
                    "begin_date": "1951-09-04",
                    "end_date": "1957-01-04",
                    "personnel_category_type_code": "V"
                }
            ]
        }
    }
}
```

## Definition of Done

- [x] Appropriate test coverage & logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)

### CODE REVIEW

**Comment by hpjaj** [comment]
@emersonthis - You'll also want to update: 

- the `service_history_request_spec.rb` 
- and the associated `service_history_response.json` 
- the Yard docs in the `show` action of the `service_histories_controller.rb`

to reflect the new attr.

**Comment by emersonthis** [comment]
@hpjaj Good catch. I updated the second two, but couldn't find anything to update in `service_history_request_spec.rb`. Those specs don't appear to reference any response attributes directly (just the json response schema)... but I am a bit mystified why the tests passed the first time.

**Comment by hpjaj** [comment]
Weird.  When you `byebug` into one of those request specs where the `service_history_response` matcher is being referenced, are you seeing the new `personnel_category_type_code` being returned from the controller's response?

**Comment by emersonthis** [comment]
It's definitely in there now. The weird part is that the specs pass even
without the property in the json schema... This shouldn't pass but it does:

```
            "service_history": {
              "description": "Array of service history hashes",
              "items": {
                "type": "hash"
              },
              "properties": {
                "branch_of_service": {
                  "type": "string"
                },
                "begin_date": {
                  "type": "string"
                },
                "end_date": {
                  "type": "string"
                }
                //,
                //"personnel_category_type_code": {
                //  "type": "string"
                //}
              },
              "type": "array"
            }
```

**Comment by emersonthis** [comment]
^ bug filed: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/11611

File: app/swagger/requests/profile.rb [hunk 1]
```diff
@@ -612,6 +612,7 @@ class Profile
                       property :branch_of_service, type: :string, example: 'Air Force'
                       property :begin_date, type: :string, format: :date, example: '2007-04-01'
                       property :end_date, type: :string, format: :date, example: '2016-06-01'
+                      property :personnel_category_type_code, type: :string, example: 'V', description: 'A = Regular Active, N = Guard, V = Reserve, Q = Reserve Retiree'
```
Comment (bshyong) [inline app/swagger/requests/profile.rb]: 💯

**Pull request merged**